### DISCUSSION TITLE
Nut Ups PowerWalker

### DISCUSSION THREAD

**Discussion by evirci** [indomus/forum#2597] [HUB: Home Assistant]
Buonasera a tutti, ho integrato la parte server NUT per integrare un UPS PowerWalker VI800SW su Home assistant OS, e sembra dare qualche segnale di vita, ma senza farmi loggare...
sapreste gentilmente spiegarmi cosa potrei cambiare affinchè possa vedere i dati? L'ups è collegato via usb b - usb a al mini pc.

```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service base-addon-banner: starting
-----------------------------------------------------------
 Add-on: Network UPS Tools
 Manage battery backup (UPS) devices
-----------------------------------------------------------
 Add-on version: 0.12.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 10.5  (amd64 / qemux86-64)
 Home Assistant Core: 2023.8.3
 Home Assistant Supervisor: 2023.08.1
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service base-addon-banner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service base-addon-timezone: starting
s6-rc: info: service base-addon-log-level: starting
s6-rc: info: service fix-attrs successfully started
[18:29:00] INFO: Configuring timezone (Europe/Rome)...
s6-rc: info: service base-addon-log-level successfully started
s6-rc: info: service base-addon-timezone successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/nut.sh
[18:29:01] INFO: Setting mode to netserver...
[18:29:02] INFO: Generating /etc/nut/upsd.users...
[18:29:02] INFO: Configuring user: nutty
[18:29:03] INFO: Password is NOT in the Have I Been Pwned database! Nice!
[18:29:05] INFO: Configuring Device named powerwalker...
[18:29:06] INFO: Starting the UPS drivers...
Using protocol: Voltronic-QS 0.07
No values for battery high/low voltages
Using 'guesstimation' (low: 10.400000, high: 13.000000)!
Battery runtime will not be calculated (runtimecal not set)
Network UPS Tools - UPS driver controller 2.8.0
cont-init: info: /etc/cont-init.d/nut.sh exited 0
cont-init: info: running /etc/cont-init.d/nutclient.sh
cont-init: info: /etc/cont-init.d/nutclient.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun upsd (no readiness notification)
services-up: info: copying legacy longrun upsmon (no readiness notification)
s6-rc: info: service legacy-services successfully started
[18:29:08] INFO: Starting the UPS information server...
   0.000000	fopen /run/nut/upsd.pid: No such file or directory
   0.000063	Could not find PID file '/run/nut/upsd.pid' to see if previous upsd instance is already running!
   0.000821	[D1] debug level is '1'
   0.001019	listening on 0.0.0.0 port 3493
   0.002093	Connected to UPS [powerwalker]: nutdrv_qx-powerwalker
   0.005065	Running as foreground process, not saving a PID file
[18:29:08] INFO: Starting the UPS monitor and shutdown controller...
   0.000000	fopen /run/nut/upsmon.pid: No such file or directory
   0.000173	Could not find PID file to see if previous upsmon instance is already running!
   0.001089	Using power down flag file /etc/killpower
   0.001817	UPS: powerwalker@localhost (primary) (power value 1)
   0.001946	[D1] debug level is '1'
   0.002034	[D1] debug level is '1'
   0.002086	Warning: running as one big root process by request (upsmon -p)
   0.002204	[D1] Saving PID 460 into /run/nut/upsmon.pid
Init SSL without certificate database
   0.007963	[D1] Trying to connect to UPS [powerwalker@localhost]
   0.036973	User upsmonmaster@127.0.0.1 logged into UPS [powerwalker]
   0.009986	[D1] Logged into UPS powerwalker@localhost
```

**Comment by lcorsini** [comment]
in che senso farti loggare?
hai installato anche upsc? lo puoi usare per testare che l'ups viene monitorato con 
`upsc powerwalker@localhost`

per integrarlo poi con Homeassistant devi creare un utente nel file /etc/upsd.upsers
`[utente_ups]
  password = PASSWORD
  upsmon master
`

**Reply by evirci** [reply]
mi sembra che non riesca ad eseguire il login dal registro che ho riportato nel primo post, sbaglio?
UPSC? è un altro componente aggiuntivo? riusciresti a spiegarmi?
Da quel log non è che ci capisca molto...

**Reply by lcorsini** [reply]
dovresti postare i file di configurazione che hai modificato, giusto per vedere la configurazione, anche per capire il driver che stai usando
dal log ti direi che riesce a connettersi all'ups `Logged into UPS powerwalker@localhost`
upc è un tool da riga di comando per interrogare un UPS configurato sotto nut.
per l'integrazione con homeassistant serve poi una utenza nel file che ti ho indicato.
se sia homeassistant che nut sono sulla stessa macchina il fatto che nut risponda su localhost va benissimo

**Reply by evirci** [reply]
Innanzitutto grazie per la risposta ed il tempo che mi stai dedicando...
Questa è la configurazione che ho impostato sulla configurazione del componente aggiuntivo Network UPS Tools.
```
users:
  - username: nutty
    password: xxxxxx
    instcmds:
      - all
    actions: []
devices:
  - name: powerwalker
    driver: blazer_usb
    desc: PowerWalker VI800SW
    port: auto
    config: []
mode: netserver
shutdown_host: false
```
Non capisco dove devo creare l'utente `[utente_ups] password = PASSWORD upsmon master`
Uso HA versione OS con VirtualBox
![Immagine 2023-08-28 174900](https://github.com/indomus/forum/assets/122167724/9cabf41c-ce53-4e32-9824-8106528f0bc1)

**Reply by lcorsini** [reply]
allora quella che mi hai copiato è un file di configurazione di homeassistant, non di nut
sto leggendo ora che usi l'immagine di homeassistantOS quindi presumo che tu l'abbia installato come integrazione, e non hai accesso diretto ai veri file di configurazione, ma ti vengono generati a partire da quello che hai postato e non puoi installare il comando da cli di cui ti parlavo.

detto questo, quindi, la parte users: corrisponde a quanto verrebbe scritto nel file di di cui ti parlavo.
dal log pare connesso, quindi sembrerebbe che l'addon sia configurato.
a questo punto dai device dovresti configurare l'intgrazione nut mettendo come host localhost e come utente quello che hai definito sotto users:

**Reply by evirci** [reply]
ed è quello che tentavo di fare, ma invano... 
![Immagine 2023-08-28 185605](https://github.com/indomus/forum/assets/122167724/138eeb97-3223-48fc-942a-7571fb1d8d34)

![Immagine 2023-08-28 185820](https://github.com/indomus/forum/assets/122167724/a7959135-f342-465a-b49f-45d45c37ac0c)

Logicamente la password è corretta...
Esatto, praticamente NUT lato server è un componente aggiuntivo, lato user bisogna aggiungere un'integrazione, che a quanto pare non vuole saperne di connettesi...

**Reply by snakuzzo** [reply]
Puoi fare una prova?
Dalla CLI di HA OS...

```
nc -zvw2 localhost 3493
```

Se il servizio è attivo deve darti `open` (o `succeded`)

es.

```
localhost [127.0.0.1] 3493 (nut) open
```

Comunque nel momento in cui provi a configurare l'integrazione, se fallisce, con il livello di log che c'è di default, su core dovrebbe indicarti il tipo di errore sul log di core appunto

**Reply by lcorsini** [reply]
probabilmente non è esposto localhost perchè la rete di docker non è configurata come host,
dalla documentazione dell'addon:

Note: The host a0d7b954-nut can be used to allow Home Assistant to communicate directly with the addon

https://github.com/hassio-addons/addon-nut/blob/main/nut/DOCS.md

**Reply by evirci** [reply]
<img width="259" alt="image" src="https://github.com/indomus/forum/assets/122167724/dcabd4b3-06b8-4738-8156-b1c6c94a739f">

Questo è il risultato, anche se non capisco perchè mi cambi la stringa nonostante metta l'esatta copia della tua...

<img width="343" alt="image" src="https://github.com/indomus/forum/assets/122167724/71bf121f-dd38-4332-b269-4500f7d6fcab">

Questo invece è l'errore sul log core di HA che compare quanto tento di configurare l'integrazione user

**Reply by snakuzzo** [reply]
> probabilmente non è esposto localhost perchè la rete di docker non è configurata come host, dalla documentazione dell'addon:
> 
> Note: The host a0d7b954-nut can be used to allow Home Assistant to communicate directly with the addon
> 
> https://github.com/hassio-addons/addon-nut/blob/main/nut/DOCS.md

Sì, potrebbe essere... anche se non mi torna. Ho provato sulla VM su cui ho l'installazione di HA OS di test e dall'host (HA OS appunto) riesco correttamente a fare un test puntando localhost e la porta interessata.
Inoltre, ho verificato anche con docker...e Core è configurato su network host

![image](https://github.com/indomus/forum/assets/7205371/2227b738-8c10-4769-be61-a14aabc22e23)

**Reply by evirci** [reply]
ho provato ad usare `a0d7b954-nut` al posto di `localhost` , restituisce sempre impossibile connettersi

**Reply by snakuzzo** [reply]
ma tu sei sicuro che l'addon sia partito correttamente o che non si riavvii di continuo ?
Puoi controllare i log dell'addon e anche quelli del supervisor ?

**Reply by evirci** [reply]
Grazie a voi, ho risolto... non mi ero accorto che nella schermata principale dell'integrazione era presente un `Nome host` ed un nome leggermente diverso da quello suggerito da lcorsini ma con la stessa sintassi...messo quello è andato al volo! Grazie mille ragazzi, sempre il top!

**Reply by snakuzzo** [reply]
Se hai risolto...a posto così :)
Anche se...
Ora non so com'è fatto questo addon, ma se  il container di core viene tirato su con network host e comunica con gli altri container anche  con localhost. Ovviamente gli altri container devono essere configurati per esporre i servizi sull'host con il bind delle porte.
Probabilmente con l'addon che stai usando questo non avviene.
